pkg/controllers/metrics/pod/controller.go

```diff
@@ -132,22 +164,57 @@ func (c *Controller) Reconcile(ctx context.Context, req reconcile.Request) (reco
 			Labels:   labels,
 		},
 	})
-	c.recordPodStartupMetric(pod)
+	c.recordPodStartupMetric(pod, labels)
 	return reconcile.Result{}, nil
 }
 
-func (c *Controller) recordPodStartupMetric(pod *corev1.Pod) {
+func (c *Controller) recordPodStartupMetric(pod *corev1.Pod, labels prometheus.Labels) {
```

Comment:
nit:This function name is starting to get misleading -- can we consider renaming it/breaking out this single function into multiple so it's easier to track what things are happening here.

Reply:
Agreed